lib/Dialect/Torch/IR/TorchOps.cpp

```diff
@@ -1494,6 +1494,15 @@ static PrimDictConstructOp getDictConstructIfNotModified(Value torchDict) {
   return torchDict.getDefiningOp<Torch::PrimDictConstructOp>();
 }
 
+static PrimListConstructOp isListConstructNotModified(Value torchList) {
+  if (!llvm::all_of(torchList.getUsers(), [](Operation *op) {
+        return isa<Aten__Contains__IntListOp>(op);
+      }))
+    return nullptr;
```

Comment:
The idea is to make sure that there is not an op between the moment the list value is created and the moment the `__contains__` op is called that overwrites the list in place. Checking that the only user is the `__contains__` op is a pretty conservative way of ensuring no mutation has taken place since the list argument was created. However, because it is very conservative, this does mean that we will miss valid cases where a folding could've taken place. For example, if an overwrite happens after the `__contains__` op, this check will prevent the fold from happening.

The structure of this function is borrowed from the one used for dict constructs, which does a slightly more comprehensive check:

https://github.com/llvm/torch-mlir/blob/e143a34948dcc845cddd8d21f5f13862520b1e0e/lib/Dialect/Torch/IR/TorchOps.cpp#L1487-L1495